pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -261,18 +263,14 @@ func (builder *ListenerBuilder) buildVirtualOutboundListener(
 		isTransparentProxy = proto.BoolTrue
 	}
 
-	fallthroughNetworkFilters := buildFallthroughNetworkFilters(push, node)
-
-	filterChains := []*listener.FilterChain{
-		{
-			Filters: fallthroughNetworkFilters,
-		},
-	}
+	var filterChains []*listener.FilterChain
 
 	// The virtual listener will handle all traffic that does not match any other listeners, and will
 	// blackhole/passthrough depending on the outbound traffic policy. When passthrough is enabled,
 	// this has the risk of triggering infinite loops when requests are sent to the pod's IP, as it will
 	// send requests to itself. To block this we add an additional filter chain before that will always blackhole.
+	// Add this as the first filter chain in the virtual outbound. Unsure of the resolution priority in Envoy
```

Comment:
The order listed here does not influence the priority, documented:

https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/listener/listener_components.proto#listener-filterchainmatch

Reply:
If I read correctly, the PR is going to to three filter chains to the outbound.
|   | Transport protocol|ALPN|
|---|---|---|
|FCM1(tcp)| tls |   |
|FCM2(tcp)|   |   |
|FCM3(http)|   | HTTP/1.1, HTTP/2  |

The filter chain order doesn't affect the priority and the filter chain match has some ordering. Considering HTTPs traffic, the TLS inspector will set the transport protocol (to `tls`), and ALPN(to `http/1.1` or sth. else). Envoy will select FCM1.